apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -216,8 +216,14 @@ get_rules_with_same_event(Topic) ->
 get_rule_ids_by_action(ActionName) when is_binary(ActionName) ->
     [
         Id
-     || #{actions := Acts, id := Id} <- get_rules(),
-        lists:any(fun(A) -> A =:= ActionName end, Acts)
+     || #{actions := Acts, id := Id, from := From} <- get_rules(),
+        lists:any(fun(A) -> A =:= ActionName end, Acts) orelse
+            lists:any(
+                fun(BridgeID) ->
+                    BridgeID =:= ActionName
+                end,
+                [BridgeID || <<"$bridges/", BridgeID/binary>> <- From]
```

Comment:
nit: there should be a function to parse resource ID to bridge ID ?

Reply:
Do you mean bridge hookpoint to bridge id?  Because the targeted bridge id is the input to this function, so we can just compare for equality when searching for dependencies.